src/Lib/Managers/HoverManager.vala

```diff
@@ -0,0 +1,144 @@
+/*
+* Copyright (c) 2019 Alecaddd (http://alecaddd.com)
+*
+* This file is part of Akira.
+*
+* Akira is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+
+* Akira is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+
+* You should have received a copy of the GNU General Public License
+* along with Akira.  If not, see <https://www.gnu.org/licenses/>.
+*
+* Authored by: Giacomo Alberini <giacomoalbe@gmail.com>
+*/
+
+public class Akira.Lib.Managers.HoverManager : Object {
+    public weak Goo.Canvas canvas { get; construct; }
+
+    private double initial_event_x;
+    private double initial_event_y;
+    private Goo.CanvasItem hover_effect;
+
+    public HoverManager (Goo.Canvas canvas) {
+        Object (
+            canvas: canvas
+        );
+    }
+
+    construct {
+    }
+
+    public void set_initial_coordinates (double event_x, double event_y) {
+        initial_event_x = event_x;
+        initial_event_y = event_y;
+    }
+
+    public void add_hover_effect (double event_x, double event_y) {
+        remove_hover_effect ();
+        set_cursor_for_nob (Managers.NobManager.Nob.NONE);
+
+        var target = canvas.get_item_at (event_x, event_y, true);
+
+        if (target == null) {
+            return;
+        }
+
+        if (target is Models.CanvasItem) {
+            var item = target as Models.CanvasItem;
+
+            double line_width = 0.0;
+            item.get ("line-width", out line_width);
+
+            // Each item is always at 0,0 relative to its
+            // coordinate system
+            double x = 0 - line_width / 2;
+            double y = 0 - line_width / 2;
+
+            double width = item.get_coords ("width");
+            double height = item.get_coords ("height");
+
+            x -= line_width / 2;
+            y -= line_width / 2;
+            width += line_width * 2;
+            height += line_width * 2;
+
+            if (!item.selected) {
+                hover_effect = new Goo.CanvasRect (
+                    null,
+                    x, y,
+                    width, height,
+                    "line-width", 2.0,
+                    "stroke-color", "#41c9fd",
+                    null
```

Comment:
Make the line-width and stroke-color values constants at the beginning of the class, so they're easier to find and update.

Reply:
Nope, just a leftover! Housecleaning necessary :)